src/context/DataContext.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+    createContext,
+    Dispatch,
+    ReactNode,
+    SetStateAction,
+    useState,
+} from 'react';
+
+import { ProductInterface } from '@models/product';
+import { AuthUser } from '@models/user';
+
+type DataContextType = {
+    user: AuthUser;
+    setUser: Dispatch<SetStateAction<AuthUser>>;
+
+    products: ProductInterface[];
+    setProducts: Dispatch<SetStateAction<ProductInterface[]>>;
+};
+
+const DataContext = createContext<DataContextType>({} as DataContextType);
+
+export const DataContextProvider: React.FC<{ children: ReactNode }> = ({
+    children,
+}: {
+    children: ReactNode;
+}) => {
```

Comment:
```suggestion
export const DataContextProvider = ({
    children,
}: {
    children: ReactNode;
}) => {
```
we can just type the props rest will be inferred. Can remove React.FC from other places as well just type the props